medcat/meta_cat.py

```diff
@@ -319,7 +355,7 @@ def save(self, save_dir_path: str) -> None:
         self.tokenizer.save(save_dir_path)
 
         # Save config
-        self.config.save(os.path.join(save_dir_path, 'config.json'))
+        # self.config.save(os.path.join(save_dir_path, 'config.json'))
```

Comment:
so we're saving config? This change shouldn't affect the existing Bi-LSTM models, or if a user wants to use the bi-LSTM implementation

Reply:
we aren't saving the config as of now, the reason is the version of the config that is being saved is drastically different to the one we are passing, causing issues when the config is again being read and modified. Here is the version being saved:

`{
  "general": {
    "py/object": "medcat.config_meta_cat.General",
    "py/state": {
      "__dict__": {
        "device": "cpu",
        "disable_component_lock": false,
        "seed": 13,
        "description": "No description",
        "category_name": "Status",
        "category_value2id": {"Other": 1, "Confirmed": 0},
        "vocab_size": 30522,
        "lowercase": true,
        "cntx_left": 20,
        "cntx_right": 10,
        "replace_center": null,
        "batch_size_eval": 5000,
        "annotate_overlapping": false,
        "tokenizer_name": "bert-tokenizer",
        "save_and_reuse_tokens": false,
        "pipe_batch_size_in_chars": 20000000,
        "span_group": null
      },
      "__fields_set__": {
        "py/set": [
          "device"                  ,
          "lowercase"               ,
          "save_and_reuse_tokens"   ,
          "seed"                    ,
          "category_value2id"       ,
          "replace_center"          ,
          "pipe_batch_size_in_chars",
          "annotate_overlapping"    ,
          "disable_component_lock"  ,
          "batch_size_eval"         ,
          "cntx_right"              ,
          "cntx_left"               ,
          "tokenizer_name"          ,
          "vocab_size"              ,
          "category_name"
        ]
      },
      "__private_attribute_values__": {}
    }
  },
  "model": {
    "py/object": "medcat.config_meta_cat.Model",
    "py/state": {
      "__dict__": {
        "model_name": "bert",
        "model_variant": "bert-base-uncased",
        "model_freeze_layers": false,
        "num_layers": 3,
        "input_size": 1536,
        "hidden_size": 32,
        "dropout": 0.25,
        "category_undersample": "Other",
        "model_architecture_config": {
          "fc2"         : true ,
          "fc3"         : false,
          "lr_scheduler": true
        },
        "num_directions": 2,
        "nclasses": 2,
        "padding_idx": 0,
        "emb_grad": true,
        "ignore_cpos": false,
        "load_model_dict_": false,
        "train_on_full_data": true
      },
      "__fields_set__": {
        "py/set": [
          "dropout"                  ,
          "ignore_cpos"              ,
          "padding_idx"              ,
          "hidden_size"              ,
          "train_on_full_data"       ,
          "model_name"               ,
          "emb_grad"                 ,
          "input_size"               ,
          "num_directions"           ,
          "nclasses"                 ,
          "model_architecture_config",
          "load_model_dict_"         ,
          "model_variant"            ,
          "model_freeze_layers"      ,
          "num_layers"               ,
          "category_undersample"
        ]
      },
      "__private_attribute_values__": {}
    }
  },
  "train": {
    "py/object": "medcat.config_meta_cat.Train",
    "py/state": {
      "__dict__": {
        "batch_size": 64,
        "nepochs": 20,
        "lr": 0.0005,
        "test_size": 0.2,
        "shuffle_data": true,
        "class_weights": [0.3, 0.65],
        "score_average": "weighted",
        "prerequisites": {},
        "cui_filter": null,
        "auto_save_model": true,
        "last_train_on": null,
        "metric": {"base": "macro avg", "score": "f1-score"},
        "loss_funct": "cross_entropy",
        "gamma": 3,
        "loss_function": "focal_loss"
      },
      "__fields_set__": {
        "py/set": [
          "batch_size"     , "lr"             , "auto_save_model",
          "class_weights"  , "nepochs"        , "test_size"      ,
          "gamma"          , "prerequisites"  , "score_average"  ,
          "loss_function"  , "cui_filter"     , "shuffle_data"
        ]
      },
      "__private_attribute_values__": {}
    }
  }
}
`

Reply:
sorry I don't understand your point of why the config isn't being saved? The modified config class needs to be backwards compatible with prev models. If we need a Bert specific implementation config there should be a new class that is used

Reply:
Fixed this, config is being saved now